fragdenstaat_de/fds_cms/templates/foirequest/cms_plugins/map.html

```diff
@@ -0,0 +1,311 @@
+                    var popupHtml = createPopupContent([
+                        {% for foirequest in group.requests %}
+                            {
+                                url: "{{ foirequest.get_absolute_url }}",
+                                title: "{{ foirequest.title|escapejs }}",
+                                public_body_name: "{{ foirequest.public_body.name|escapejs }}",
+                                status_display: "{{ foirequest.get_status_display|escapejs }}"
+                            }{% if not forloop.last %},{% endif %}
+                        {% endfor %}
+                    ]);
+
+                    var polygon = L.geoJSON(regionCoords, {
+                        style: {
+                            color: getStatusIcon('{{ group.requests.0.status }}').options.html.match(/background-color: ([^;]+)/)[1],
+                            weight: 2,
+                            opacity: 0.8,
+                            fillOpacity: 0.2
+                        }
+                    }).addTo(map);
+                    polygon.bindPopup(popupHtml);
+                    bounds.extend(polygon.getBounds());
+                    polygons.push(polygon);
+                } catch (e) {
+                    console.error('Error creating polygon:', e);
+                }
+            }
+        {% endfor %}
+
+        // Fit map to bounds if we have any markers or polygons
+        if (markers.length > 0 || polygons.length > 0) {
+            var group = new L.featureGroup(markers.concat(polygons));
+            map.fitBounds(bounds);
+        }
+
+        // Add legend
+        var legend = L.control({position: 'bottomright'});
+        legend.onAdd = function (map) {
+            var div = L.DomUtil.create('div', 'foirequest-legend');
+            div.innerHTML += '<b>{% trans "Status" %}</b><br>';
+            div.innerHTML += '<i style="background:#ffc107"></i> {% trans "Pending" %}<br>';
+            div.innerHTML += '<i style="background:#28a745"></i> {% trans "Resolved" %}<br>';
+            div.innerHTML += '<i style="background:#dc3545"></i> {% trans "Refused" %}<br>';
+            div.innerHTML += '<i style="background:#17a2b8"></i> {% trans "Partially successful" %}<br>';
+            div.innerHTML += '<i style="background:#fd7e14"></i> {% trans "Overdue" %}<br>';
+            div.innerHTML += '<i style="background:#6c757d"></i> {% trans "Asleep" %}<br>';
+            {% if requests_without_geo > 0 %}
+            div.innerHTML += '<hr class="legend-divider">';
+            div.innerHTML += '<div class="legend-note">';
+            div.innerHTML += '&#9432; {% blocktrans count counter=requests_without_geo %}{{ counter }} request without geo data{% plural %}{{ counter }} requests without geo data{% endblocktrans %}';
+            div.innerHTML += '</div>';
+            {% endif %}
+            return div;
+        };
+        legend.addTo(map);
+    }
+</script>
+<style>
+.foirequest-map-container {
```

Comment:
Let's also move this CSS into its own CSS file. Using the `addtoblock` template tag will also render only one load of the CSS which is a nice bonus.